src/test/browserutils.ts

```diff
@@ -0,0 +1,60 @@
+import puppeteer from 'puppeteer';
+
+// jest-puppeteer will expose the `page` and `browser` globals to Jest tests.
+declare global {
+  var page: puppeteer.Page;
+  var browser: puppeteer.Browser;
+}
+
+if (typeof(browser) === 'undefined' || typeof(page) === 'undefined') {
+  throw new Error('Browser has not been started! Did you remember to specify `@jest-environment puppeteer`?');
+}
+
+const ignoreConsoleMessages = [
+  '%cDownload the React DevTools for a better development experience: https://fb.me/react-devtools font-weight:bold',
+  '%cHowdy! If you want to help out, the source code can be found at https://github.com/openstax/books-web font-weight:bold', // tslint:disable-line:max-line-length
+];
+
+page.on('console', (consoleMessage) => {
+  const text = consoleMessage.text();
+  if (ignoreConsoleMessages.indexOf(text) === -1) {
+    console.log(text); // tslint:disable-line:no-console
+  }
+});
+
+const DEV_SERVER_PORT = 8000;
+
+export const url = (path: string) => `http://localhost:${DEV_SERVER_PORT}/${path.replace(/^\/+/, '')}`;
+
+export const navigate = async(target: puppeteer.Page, path: string) => {
+  await target.goto(url(path));
+
+  await target.evaluate(async() => {
+    if (window) {
+      await window.__APP_ASYNC_HOOKS.calm();
+    }
+  });
+};
+
+export const finishRender = async(_: puppeteer.Page) => {
+  // HACK - there is no convenient way to tell if chrome is finished rendering,
+  // we should investigate inconvenient possibilities.
```

Comment:
@philschatz the problem is:

i've invented ways to be sure that the relevant data and network requests for fonts have been loaded, but i can't tell when chrome is finished drawing the page. this is irrelevant for some browser tests because the interactions will already work, but for screenshots there are problems like:

- [image](https://jenkins-dev.cnx.org:8080/job/openstax-continuous-integration/job/rex-web/job/PR-15/49/artifact/content-browserspec-ts-content-looks-right-1-diff.png) the page just isn't drawn at the bottom, adding the 1s delay fixes this issue so i'm assuming its just draw time.
- fonts rendering is different from the network request being done, there is no great way to tell when fonts are done loading. `document.fonts.ready` is a thing, but doesn't seem reliable yet (very experimental).

i don't have any really good ideas, maybe keep taking screenshots every 50ms or so and see when they stop changing? i feel like that might give false positives if the rendering happens in chunks. do you have any thoughts? 

we can merge this as is and circle back if necessary, it seems to be passing find on jenkins